Site_project/cart/views.py

```diff
@@ -0,0 +1,48 @@
+from django.contrib.auth.decorators import login_required
+from django.shortcuts import render, redirect, get_object_or_404
+from django.urls import reverse_lazy, reverse
+
+from catalog.views import HomeView
+from .models import Cart, CartProduct, Order, Customer
+from catalog.models import Product
+
+
+@login_required
+def add(request, product_slug):
+    slug_product = Product.objects.get(slug=product_slug)
```

Comment:
```suggestion
    product = Product.objects.get(slug=product_slug)
```
И вопрос: а что если товара нет?